web-ui/src/pages/CheckinsReportPage.jsx

```diff
@@ -120,12 +121,16 @@ const CheckinsReportPage = () => {
     }
   }, [processedQPs.current]);
 
-  // Set the mapped PDLs to the PDLs with members
+  // Set the mapped PDLs to the PDLs with members attached
+  // filtering out data about check-ins under a different PDL.
   useEffect(() => {
     if (!pdls) return;
-    pdls.forEach(
-      pdl => (pdl.members = selectTeamMembersWithCheckinPDL(state, pdl.id))
-    );
+    pdls.forEach(pdl => {
```

Comment:
Does the code in the function passed to `forEach` cause some kind of side-effect?
Maybe the calls to `selectTeamMembersWithCheckinPDL` are necessary, but I don't understand why the `filter` call is necessary or the `return` keyword.